### DISCUSSION TITLE
Cannot expand attributes of int subclass in Variables, Watch or Debug Console

### DISCUSSION THREAD

**Discussion by laundmo** [microsoft/debugpy#1758] [Enhancements]
## Problem

Subclasses of int (and other "primitives") cannot be expanded and their attributes cannot be inspected.

## Reproduction

Debug this python file:
```py
class SomeIntSubclass(int):
    def some_method(self):
        pass


class SomeClass:
    def some_method(self):
        pass


normal_instance = SomeClass()
int_instance = SomeIntSubclass(1)
breakpoint()
```

As you can see, there's no toggle to expand/collapse the attributes list
![Image](https://github.com/user-attachments/assets/7119a785-4630-49f0-9d68-c7b77f4084e1)
![Image](https://github.com/user-attachments/assets/deb8c2f0-2336-4b08-897f-99303926cf72)

## Solutions
I've orderd solutions i can think of in order of preference
- Instead of using `isinstance(x, int)` or a equivalent to check whether a variable is a primitive type, use `type(x) is int` so only subclasses show up as normal instances.
- A setting to treat all values, be they "primitives" or not, as objects which can be inspected
- A special cased command, function, or similar in the Debug Console to inspect a "primitive" as if it was a normal instance

**Comment by rchiodo** [comment]
Thanks for the suggestion. I believe this would need some sort of setting

Internally debugpy just does a `__repr__` on most variables. That code is here:

https://github.com/microsoft/debugpy/blob/5014f2538ac04e90e1435f8f2533a20a3b6c29fa/src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_safe_repr.py#L12.

Transferring to a discussion item for upvotes.

**Reply by laundmo** [reply]
I had not thought of this as a feature suggestion as it seems to be a bug in how debugpy, pydevd, or whatever actually does this decides whether to display a simple repr or the expandable inspection thingie.

Does this move to discussions mean debugpy devs don't consider this behaviour a bug?
